dotCMS/src/main/java/com/dotcms/security/apps/SecretCachedKeyStoreImpl.java

```diff
@@ -0,0 +1,204 @@
+package com.dotcms.security.apps;
+
+import static com.dotcms.security.apps.AppsCache.CACHE_404;
+import static com.dotcms.security.apps.AppsUtil.digest;
+import static com.dotcms.security.apps.SecretsKeyStoreHelper.SECRETS_KEYSTORE_PASSWORD_KEY;
+
+import com.dotcms.enterprise.cluster.ClusterFactory;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableList;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * This is basically a safe repository implemented using java.security.KeyStore
+ * Which according to the official Java documentation Represents a storage facility for cryptographic keys and certificates.
+ * This Class serves as the entry point to the repository whose implantation really resides in SecretsKeyStoreHelper
+ * This class adds caching
+ */
+public class SecretCachedKeyStoreImpl implements SecretsStore {
+
+    private final SecretsKeyStoreHelper secretsKeyStore;
+    private final AppsCache cache;
+
+    SecretCachedKeyStoreImpl(final AppsCache cache) {
+        this.secretsKeyStore = new SecretsKeyStoreHelper(() -> cache
+                .getFromCache(SECRETS_KEYSTORE_PASSWORD_KEY, () -> Config
+                        .getStringProperty(SECRETS_KEYSTORE_PASSWORD_KEY,
+                                digest(ClusterFactory.getClusterSalt())).toCharArray()),
+                ImmutableList.of(this::flushCache));
+        this.cache = cache;
+    }
+
+    public SecretCachedKeyStoreImpl() {
+       this(CacheLocator.getAppsCache());
+    }
+
+    /**
+     * Verifies if the key exists in the store
+     * @param variableKey
+     * @return
+     */
+    @Override
+    public boolean containsKey(final String variableKey) {
+        try {
+            return getKeysFromCache().contains(variableKey);
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)